src/child-applications/lifecycles/load.js

```diff
@@ -13,7 +13,12 @@ export async function toLoadPromise(app) {
 	let appOpts;
 
 	try {
-		appOpts = await app.loadImpl();
+		const loadPromise = app.loadImpl();
+		if (!smellsLikeAPromise(loadPromise)) {
+			// The name of the child app will be prepended to this error message inside of the handleChildAppError function
+			throw new Error(`single-spa loading function did not return a promise. Check the second argument to declareChildApplication('${app.name}', loadingFunction, activityFunction)`);
```

Comment:
Here's the useful error message for if a loading function is incorrectly implemented.